cyences_app_for_splunk/bin/sophos_instance_details.py

```diff
@@ -0,0 +1,217 @@
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        list_of_uuid.append(response_json)
+                        return list_of_uuid
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            elif(hostname!="" and ip!="" and hostname!=" " and ip!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname+"ipAddresses="+ip, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            elif(ip!="" and ip!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?ipAddresses="+ip, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+
+            elif(hostname!="" and hostname!=" "):
+                for tenant in SOPHOS_TENANT_DICT:
+                    requestHeaders = {
+                        "X-Tenant-ID": tenant,
+                        "Authorization": "Bearer "+ barier_token,
+                        "Accept": "application/json"
+                    }
+                    response = requests.get(SOPHOS_TENANT_DICT[tenant]+"/endpoint/v1/endpoints?hostnameContains="+hostname, headers=requestHeaders)
+                    response_json = response.json()
+                    if response.status_code == 200:
+                        for instance in response_json.get("items"):
+                            list_of_uuid.append(instance)
+                    else:
+                        self.logger.error("Error while fetching UUID of instance")
+
+            return list_of_uuid
+        except Exception as e:
+            self.logger.error("Error while fetching Instance UUID : {}".format(e))
```

Comment:
instead use 
`logger.exception("Error while fetching Instance UUID")`
to also add stacktrace in the logs.